### PR TITLE
chore: backport TypeAlias for py3.9 compat

### PR DESCRIPTION
**Pull request by stillmatic**

### CODE REVIEW

**Review by stillmatic — COMMENTED** [review]
see commentary here: https://github.com/lstein/stable-diffusion/issues/302#issuecomment-1234917170

**Comment by magnusviri** [comment]
@Birch-san the lstein repo isn't working because of this. https://github.com/lstein/stable-diffusion/issues/335

**Comment by magnusviri** [comment]
@Birch-san yesterday somebody downgraded the lstein mac environment to python 3.9, that's why it crashes now. This pull request makes this repo work with 3.9.

I'm not sure if we will leave it at 3.9. They are [discussing it here](https://github.com/lstein/stable-diffusion/discussions/331).

**Comment by stillmatic** [comment]
I would note that this change is compatible with 3.9 and 3.10, so if this change is merged, you can use whichever version you please

**Comment by Birch-san** [comment]
thanks! I'll integrate this. I'll also rebase onto latest (Katherine added a trivial optimization https://github.com/crowsonkb/k-diffusion/commit/4567328deec0f90a549dde53c1858cf7ed0140dd).

I think I can simplify `clone_please` to be a more sensible patch, now that I found the source of the ±inf problem.  
https://github.com/pytorch/pytorch/issues/84364  
just need to `.detach().clone()` the result returned from `utils.py::append_dims`.

Unrelated to MPS, but the following two points pertain to the patches I've made for getting good results at low sample steps (e.g. 7 or 8)…  
https://twitter.com/Birchlabs/status/1565029734865584143?s=20&t=lU-1bZCq0IejmzGM3YzVrg

I think I'll delete the `decorate_sigma_hat` option I added to the samplers (there's already an idiomatic way to do quantization in k-diffusion: `CompVisDenoiser(model, quantize=True)` — admittedly it doesn't work when churn is enabled, but nobody's using churn).  
https://github.com/crowsonkb/k-diffusion/pull/23#issuecomment-1232918456  
I anticipate that k-diffusion will update soon with an official way to quantize sigma_hat, which will mean we can get time-step discretization even when `churn>0`:  
https://github.com/crowsonkb/k-diffusion/pull/23#issuecomment-1234901445

likewise I'll delete the `concat_zero` option I added to the Karras noise schedule (it's better to increase sigma_min than to remove the zero). the same custom noise schedule of mine can be achieved without any changes to k-diffusion.  
https://github.com/crowsonkb/k-diffusion/pull/23#issuecomment-1234872495  
https://github.com/Birch-san/stable-diffusion/commit/50476fa7006aa59a939897d03f6cb8c9c746df61

**Comment by Birch-san** [comment]
oh, `TypeAlias` was only being used _because_ of my `decorate_sigma_hat` patch, which is unrelated to MPS and which I'll be removing (as explained above).

so, we don't need to force python 3.9 users to install typing-extensions. I can just delete `TypeAlias`.

`TypeAlias` wasn't that important anyway (more of a nice-to-have best-practice); the type inference would do the right thing even if I hadn't explicitly declared `TensorOperator` as a `TypeAlias`.

so I think what I'll do is delete the offending code instead of integrating this PR, if that's okay with you?

**Comment by stillmatic** [comment]
sgtm, the easiest way to fix code is delete code!

**Comment by Birch-san** [comment]
thanks 🙂 I've now force-pushed a simpler branch to `mps`, which has _no_ changes to `sampling.py`:  
https://github.com/crowsonkb/k-diffusion/compare/master...Birch-san:k-diffusion:mps

this removes the `TypeAlias` and therefore the need for this py3.9 mitigation.

anybody who's currently got my branch checked out, will not be able to pull because I force-pushed a new history. to get yourself up-to-date, cd into the k-diffusion repository (e.g. `stable-diffusion/src/k-diffusion`), fetch the latest `mps` branch and hard-reset your local mps to match the remote one:

```bash
git fetch origin mps
git reset --hard origin/mps
```
